### DISCUSSION TITLE
Update the sidebar dynamically

### DISCUSSION THREAD

**Discussion by KomoriDev** [vuejs/vitepress#4109] [Q&A]
To fullfill different users needs, I need to configure two sidebars in the document and use button to toggle. i have provided a toggle button upon the sidebar, but i don't know how can i switch sidebar dynamically

Perhaps it would be clearer with a picture
![image](https://github.com/user-attachments/assets/f290e9e5-af9e-429b-a5f2-dab51b23a4cd)

I found similar needs in the community, but it was just about generating a sidebar.

**Comment by sharpchen** [comment]
If there's a way to reassign sidebar items at runtime, you can store your sidebars using [Build-time data loading](https://vitepress.dev/guide/data-loading#build-time-data-loading) and then load it in your logic. But I don't see a api to do that, they're readonly now. See [useSidebar](https://vitepress.dev/reference/default-theme-sidebar#usesidebar)

**Reply by KomoriDev** [reply]
I got it. Thank you.
